lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.spec.ts

```diff
@@ -0,0 +1,186 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { TranslateModule } from '@ngx-translate/core';
+import { ContentTestingModule } from '../../../testing/content.testing.module';
+import { SearchChipInputComponent } from '../search-chip-input/search-chip-input.component';
+import { LogicalSearchCondition, LogicalSearchFields, SearchLogicalFilterComponent } from './search-logical-filter.component';
+
+describe('SearchLogicalFilterComponent', () => {
+    let component: SearchLogicalFilterComponent;
+    let fixture: ComponentFixture<SearchLogicalFilterComponent>;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            declarations: [SearchLogicalFilterComponent, SearchChipInputComponent],
+            imports: [
+                TranslateModule.forRoot(),
+                ContentTestingModule
+            ]
+        });
+
+        fixture = TestBed.createComponent(SearchLogicalFilterComponent);
+        component = fixture.componentInstance;
+        component.id = 'logic';
+        component.context = {
+            queryFragments: {
+                logic: ''
+            },
+            update: () => {}
+        } as any;
+        component.settings = { field: 'field1,field2', allowUpdateOnChange: true, hideDefaultAction: false };
+        fixture.detectChanges();
+    });
+
+    function getChipInputs(): HTMLInputElement[] {
+        return fixture.debugElement.queryAll(By.css('adf-search-chip-input input')).map((input) => input.nativeElement);
+    }
+
+    function getChipInputsLabels(): string[] {
+        return fixture.debugElement.queryAll(By.css('adf-search-chip-input mat-label')).map((label) => label.nativeElement.innerText);
+    }
+
+    function enterNewPhrase(value: string, index: number) {
+        const inputs = getChipInputs();
+        inputs[index].value = value;
+        fixture.detectChanges();
+        inputs[index].dispatchEvent(new KeyboardEvent('keydown', {keyCode: 13}));
```

Comment:
keycode - deprecated

Reply:
If you have the same problem like above with replacing this then see my comment: https://github.com/Alfresco/alfresco-ng2-components/pull/8616#discussion_r1214284448. We can leave it then as it is

Reply:
Yeah I tried it but it's exactly the same case